src/HAEntity.hpp

```diff
@@ -6,16 +6,27 @@
 #include <iostream>
 #include <string>
 #include <sstream>
+#include "ext/magic_enum/include/magic_enum/magic_enum_all.hpp"
 
 using std::map;
 using std::string;
 
 using json = nlohmann::json;
 
+// TODO: we need to generate this to keep this in sync with HA.
+enum class EntityType
+{
+  light,
+  Switch,
+  climate,
```

Comment:
```suggestion
  Light,
  Switch,
  Climate,
```

let's keep the capitalisation that HA uses

Reply:
This was before i discovered that cast_enum had an option for case insenitive and my editor complained about the word 'switch'. Will fix.